internal/pkg/addon/addons.go

```diff
@@ -93,10 +122,19 @@ func (a *Addons) Template() (string, error) {
 			return "", err
 		}
 	}
+
+	if a.UploadAssets {
+		mergedTemplate, err = a.packageLocalArtifacts(mergedTemplate)
+		if err != nil {
+			return "", fmt.Errorf("package local artifacts: %w", err)
```

Comment:
why not though 🥺

Reply:
Well it is to prevent the error wrapping becoming too verbose. However, it is not a must thing. In some circumstance it makes more sense to wrap errors from private functions to provide more context (but usually the inner problem is the package is too ambiguous and everything coupled tightly together)

Reply:
problem is if one of the errors isn't wrapped in the chain, then you can't call `errors.As/Is` any errors underneath that. but you're right - unless we want users (of the package) to be able to discriminate on implementation errors from this package, we shouldn't wrap. it would lead to tighter coupling between this package and someone using it. i'll remove it 😊